nixos/modules/services/games/asf.nix

```diff
@@ -0,0 +1,200 @@
+          };
+          enabled = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Whether to enable the bot on startup.";
+          };
+          settings = mkOption {
+            type = types.attrs;
+            description =
+              "Additional settings that are documented here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#bot-config";
+            default = { };
+          };
+        };
+      });
+      description = ''
+        Bots name and configuration
+      '';
+      example = {
+        exampleBot = {
+          username = "alice";
+          passwordFile = "/my/super/secret/password";
+          settings = { SteamParentalCode = "1234"; };
+        };
+      };
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users = {
+      users.asf = {
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+        group = "asf";
+        description = "Archis-Steam-Farm service user";
+      };
+      groups.asf = { };
+    };
+
+    systemd.services = {
+      asf = {
+        description = "Archis-Steam-Farm Service";
+        after = [ "network.target" ];
+
+        serviceConfig = {
+          User = "asf";
+          Group = "asf";
+          WorkingDirectory = cfg.dataDir;
+          Type = "simple";
+          ExecStart =
+            "${cfg.package}/bin/ArchiSteamFarm --path ${cfg.dataDir} --process-required --no-restart --service ";
+
+          # mostly copied from the default systemd service
+          PrivateTmp = true;
+          LockPersonality = true;
+          PrivateDevices = true;
+          PrivateIPC = true;
+          PrivateMounts = true;
+          PrivateUsers = true;
+          ProtectClock = true;
+          ProtectControlGroups = true;
+          ProtectHostname = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          ProtectProc = "invisible";
+          ProtectSystem = "full";
+          RemoveIPC = true;
+          RestrictAddressFamilies = "AF_INET AF_INET6";
+          RestrictNamespaces = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+        };
+
+        preStart = ''
+          set -x
+          # we remove config to have no complexity when creating the required files/directories
+          mkdir config || true
+          rm -rf config/*.json
+          rm -rf www
+          ln -s ${asf-config} config/ASF.json
+          echo -e '${
+            lib.strings.concatStringsSep "\\n"
+            (attrsets.mapAttrsToList mkBot cfg.bots)
+          }' \
+          | while read -r line; do
+            name="$(${pkgs.jq}/bin/jq -r .name <<< "$line")"
+            ${pkgs.jq}/bin/jq -r "del(.name)" <<< "$line" > "config/''${name}".json
+          done
```

Comment:
Can you please comment what/why is going on here? Can't we just generate files at build time instead of using `jq` to parse at runtime?

Reply:
In the beginning I had to read the files to get the password and then substitute them in the config file. Upstream was so nice and added the password file method, at my request. I'll see what I can do.

Reply:
You got upstream to add a password file option?? I can't give you enough bonus points!! You rock! 🚀 🎉

Reply:
@aanderse It benefits all users, and I'm always happy to further enhance the program if possible. Actually there was like one or two other things that I was happy to take care of, for example runtime-evaluation of `~` in `ASF_PATH` variable, it's also used in your package :slightly_smiling_face:.

Reply:
Amazing! Thanks.